packages/fxa-payments-server/src/components/NewUserEmailForm/index.tsx

```diff
@@ -170,9 +177,12 @@ export async function emailInputValidationAndAccountCheck(
   if (isEmailValid(value)) {
     valid = true;
     setValidEmail(value);
-    const response = await checkAccountExists(value);
-    hasAccount = response.exists;
-    setAccountExists(response.exists);
+    const response = await debouncedCheckAccountExists(value);
+    if (response?.exists) {
+      hasAccount = response.exists;
+      setAccountExists(hasAccount);
+    }
+    setAccountExists(hasAccount);
```

Comment:
I don't understand what's going on here.  `setAccountExists` is called both in the `if` and out of the `if`.  In that case, could simply call it outside of the `if`?    But why do we need the if in the first place?

Reply:
The `if` statement is here to avoid the error:
```
Uncaught (in promise) TypeError: can't access property "exists", response is undefined
```
I just pushed an update to remove the unnecessary call to `setAccountExists` within the `if` statement.

Reply:
How do I get this error?

Reply:
Start typing into the first input, when it gets to the time to input the tld the error will be in the console

Reply:
The error does not happen with my uodate

Reply:
> 
> 
> The error does not happen with my uodate

:excellent:  Then let's remove that `if`.

Reply:
The `if` is necessary. That is what is stopping the error from happening

Reply:
> 
> 
> > The error does not happen with my uodate
> 
> :excellent: Then let's remove that `if`.

I misunderstood what the update in the original message meant.